pkg/k8s/watchers/cilium_node.go

```diff
@@ -69,7 +69,8 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient client.Clientset, asyncContro
 							valid = true
 							isLocal := k8s.IsLocalCiliumNode(ciliumNode)
 							if oldCN.DeepEqual(ciliumNode) &&
-								comparator.MapStringEquals(oldCN.ObjectMeta.Labels, ciliumNode.ObjectMeta.Labels) {
+								comparator.MapStringEquals(oldCN.ObjectMeta.Labels, ciliumNode.ObjectMeta.Labels) &&
+								comparator.MapStringEquals(oldCN.ObjectMeta.Annotations, ciliumNode.ObjectMeta.Annotations) {
```

Comment:
This causes issue in case the wireguard key changes but the rest of the node object not
https://github.com/cilium/cilium/blob/e52fe1d59d1c6f51a76993641ea9412e0f44f749/pkg/nodediscovery/nodediscovery.go#L460-L465

since, we don't compare Annotations the event is dropped and not processed